dist/cjs/trustedtypes.api_only.js

```diff
@@ -278,6 +434,91 @@ const trustedTypesBuilderTestOnly = function() {
     return exposedPolicies.get(pName) || null;
   }
 
+  /**
+   * Returns the name of the trusted type required for a given element
+   *   attribute.
+   * @param {string} tagName The name of the tag of the element.
+   * @param {string} attribute The name of the attribute.
+   * @param {string=} elementNs Element namespace.
+   * @param {string=} attributeNs The attribute namespace.
+   * @return {string|undefined} Required type name or undefined, if a Trusted
+   *   Type is not required.
+   */
+  function getAttributeType(tagName, attribute, elementNs = '',
+      attributeNs = '') {
+    // TODO: Support namespaces.
```

Comment:
It looks like namespace support should probably go into getTypeInternal_.

Ok, so for `<svg:a xlink:href>` you have two distinct namespaces in play.
Are namespaces URLs, or common prefixes?

I wonder whether it'd be easier to code-generate a flat table that maps strings like '*/*/*/attr' would mean any element namespace, any element, any attribute namespace, attribute named attr.

Then, for given values we could cascade to different properties with fewer object, own property checks.  So for `<svg:a xlink:href>` we might try some of (I'm not sure exactly which) in order

1. `"svg/a/xlink/href"`
1. `"*/a/xlink/href"`
1. `"*/*/xlink/href"`
1. `"svg/a/*/href"`
1. `"*/a/*/href"`
1. `"*/*/*/href"`

and avoid a bunch of code that pulls out objects, tests whether they're not null and keeps looking inside.

Reply:
We probably can't use prefixes, this should be based on namespace URIs, which makes it a bit cumbersome to work with. Either the lookup key is not a single string, or we use a separator that cannot be present in a URL, like ` ` . 
The goal here is to end up with the API shape that's easiest to work with when creating DOM (and then you either have a tag name, attribute name, or the respective nodes already). Thankfully [`namespaceURI`](https://developer.mozilla.org/en-US/docs/Web/API/Node/namespaceURI) for both attribute and element nodes returns the canonical URL.

FWIW in current Chrome/Firefox in non-pure-XML documents it seems that changing the prefix of a namespace via `xmlns` makes them lose their meaning (check http://html5sec.org/test/#87 and try to rename xlink), which is reflected by `namespaceURI` returning null. It seems like some of the prefixes are simply hardcoded for X/HTML documents, that's probably specced somewhere. Nevertheless, in pure `text/html` documents namespaces can be effectively renamed, so we cannot rely on a prefix.

As a sidenote, we might be able to just avoid the problem altogether by making sure xlink just stops being risky, period. It's already deprecated: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href

Reply:
I implemented the namespaced attribute enforcement. PTAL.